src/state-prototypes.js

```diff
@@ -16,11 +16,69 @@ const StringEnum = class {
 
 const rowOverflowEnum = new StringEnum('VISIBLE', 'SCROLL', 'HIDDEN')
 // const rowTypeEnum = new StringEnum('input', 'output')
-// const cellTypeEnum = new StringEnum(
-// 'javascript', 'markdown', 'raw', 'css', 'external dependencies')
+const cellTypeEnum = new StringEnum(
+  'javascript',
+  'markdown',
+  'raw',
+  'css',
+  'external dependencies',
+)
 // const appViewEnum = new StringEnum('EXPLORE', 'REPORT') //was: 'editor', 'presentation'
 // const appModeEnum = new StringEnum('COMMAND', 'EDIT', 'TITLE', 'MENU')
 
+
+const cellSchema = {
+  type: 'object',
+  properties: {
+    content: { type: 'string' }, // change to string with default '' or 'untitled'
+    id: { type: 'integer', minimum: 0 },
+    cellType: {
+      type: 'string',
+      enum: cellTypeEnum.values(),
+    },
+    value: {}, // empty schema, `value` can be anything
+    rendered: { type: 'boolean' },
+    selected: { type: 'boolean' },
+    executionStatus: { type: 'string' },
+    evalStatus: {}, // FIXME change to string ONLY
+    rowSettings: { type: 'object' },
+  },
+  additionalProperties: false,
+}
+// cellSchema.required = Object.keys(cellSchema.properties)
+// cellSchema.minProperties = Object.keys(cellSchema.properties).length
+
+const stateSchema = {
+  type: 'object',
+  properties: {
+    title: {}, // FIXME change to string ONLY with default '' or 'untitled'
```

Comment:
Is there a reason this is an object by default instead of `null` or `undefined`?

Reply:
yea it turns out that JSON schema doesn't have "undefined", so you can use a raw object to represent the union of all types, in this case "undefined" and "string".

I'll follow up with a series of PRs that change everything in state to use single fixed types rather than type unions.